### PR TITLE
TAS-26/sign up

### PR DESCRIPTION
**Pull request by marjan-ara**

### CODE REVIEW

File: src/modules/General/components/FileUploader/index.module.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+.container {
+    width: 100%;
+    height: 126px;
+    border-radius: 12px;
+    border-radius: $radius-xl;
+    padding: 16px 24px;
+    display: flex;
+    flex-direction: column;
+    gap: 4px;
```
Comment (satanmourner) [inline src/modules/General/components/FileUploader/index.module.scss:11]: rem please

File: src/modules/General/components/FileUploader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { useFileUploader } from './useFileUploader';
+
+export const FileUploader: React.FC<FileUploaderProps> = ({
+  customStyle,
+  fileTypes,
+  maxFileSize = 10,
+  maxFileNumbers = 1,
+  setAttachments,
+}) => {
+  const { t } = useTranslation();
```
Comment (satanmourner) [inline src/modules/General/components/FileUploader/index.tsx:18]: `t` is not a clear word can you replace it with translate?

File: src/modules/General/components/FileUploader/useFileUploader.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import { useTranslation } from 'react-i18next';
+import { uploadMediaAdaptor } from 'src/core/adaptors/media';
+
+export const useFileUploader = (
+  fileTypes: string[],
+  maxFileNumbers: number,
+  maxSize: number,
+  setAttachments: (newVal: string[]) => void,
+) => {
+  const { t } = useTranslation();
```
Comment (satanmourner) [inline src/modules/General/components/FileUploader/useFileUploader.tsx:12]: it is good to rename everywhere (replace it with `translate`)

File: src/modules/General/components/FileUploader/useFileUploader.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      }
+    });
+  }, []);
+
+  const { getRootProps, getInputProps } = useDropzone({
+    onDrop,
+    accept: getAcceptedFileTypes(),
+    maxFiles: maxFileNumbers,
+  });
+  // const { getRootProps, getInputProps } = useDropzone({ onDrop });
```
Comment (satanmourner) [inline src/modules/General/components/FileUploader/useFileUploader.tsx:69]: remove this comment if you don't use it

File: src/modules/SignUp/FormHeader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
Comment (satanmourner) [inline src/modules/SignUp/FormHeader/index.tsx:1]: FormHeader move it into `SignUp/components/FormHeader`

File: src/modules/SignUp/StepperLayout/index.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
```
Comment (satanmourner) [inline src/modules/SignUp/StepperLayout/index.tsx:1]: StepperLayout move it to `SignUp/components/StepperLayout`

File: src/modules/SignUp/StepperLayout/index.module.scss [hunk 7]
```diff
@@ -0,0 +1,43 @@
+@import 'src/styles/constants/_primitives.scss';
+
+.container {
+    display: flex;
+    width: 100%;
+    height: 100vh;
+}
+
+.vertical-stepper {
```
Comment (satanmourner) [inline src/modules/SignUp/StepperLayout/index.module.scss:9]: follow BEM. you can use `.stepper`

File: src/modules/SignUp/StepperLayout/index.module.scss [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    width: 440px;
```
Comment (satanmourner) [inline src/modules/SignUp/StepperLayout/index.module.scss:10]: rem please

File: src/modules/SignUp/StepperLayout/index.module.scss [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    min-width: 440px;
+    flex-direction: column;
+    gap: 80px;
```
Comment (satanmourner) [inline src/modules/SignUp/StepperLayout/index.module.scss:13]: rem please

File: src/modules/SignUp/StepperLayout/index.module.scss [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+.form {
+    flex-grow: 1;
+    padding: 6rem 2rem;
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    align-items: center;
+
+    @media (max-width: 640px) {
```
Comment (satanmourner) [inline src/modules/SignUp/StepperLayout/index.module.scss:40]: we don't have `640px` in responsive sizes (talwind). you should use `$md` (768px)

File: src/modules/SignUp/StepperLayout/index.tsx [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+import css from './index.module.scss';
+import { StepperLayoutProps } from './index.types';
+import { useStepperLayout } from './useStepperLayout';
+
+const StepperLayout: React.FC<StepperLayoutProps> = ({ activeStep, children }) => {
+  const { steps } = useStepperLayout();
+  return (
+    <div className={css['container']}>
+      <div className={`hidden md:flex ${css['vertical-stepper']}`}>
+        <img src={'/images/logo/logo.svg'} alt="SHIN_LOGO" width="79px" height="32px" />
```
Comment (satanmourner) [inline src/modules/SignUp/StepperLayout/index.tsx:15]: Do not use direct image for logo, import it from `assets` please

File: src/pages/signUp/detail/index.module.scss [hunk 12]
```diff
@@ -0,0 +1,23 @@
+.container {
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+    gap: 2rem;
+
+    @media (min-width: 640px) {
```
Comment (satanmourner) [inline src/pages/signUp/detail/index.module.scss:7]: use `$md` please

File: src/pages/signUp/detail/index.module.scss [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+        max-width: 480px;
```
Comment (satanmourner) [inline src/pages/signUp/detail/index.module.scss:8]: rem please

File: src/pages/signUp/email/index.module.scss [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+        background-repeat: no-repeat;
+        background-size: cover;
+    }
+}
+
+.formSection {
+    flex-grow: 1;
+    display: flex;
+    flex-direction: column;
+    padding: 32px;
```
Comment (satanmourner) [inline src/pages/signUp/email/index.module.scss:43]: rem please

Comment (satanmourner) [reply src/pages/signUp/email/index.module.scss:43]: 54, 50 line too

File: src/pages/signUp/email/index.module.scss [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+    display: flex;
+}
+
+.section {
+    flex-direction: column;
+    gap: 6rem;
+    padding-top: 6rem;
+    padding-left: 5rem;
+    background-color: $color-grey-100;
+    max-width: 800px;
```
Comment (satanmourner) [inline src/pages/signUp/email/index.module.scss:15]: rem please

File: src/pages/signUp/email/index.tsx [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+import { useEmail } from './useEmail';
+
+export const Email = () => {
+  const { register, errors, handleSubmit, onContinue } = useEmail();
+  const { t } = useTranslation();
+
+  return (
+    <div className={css['container']}>
+      <div className={`w-full lg:min-w-[600px] ${css['formSection']}`}>
+        <img src={'/images/logo/logo.svg'} alt="SHIN_LOGO" width="79px" height="32px" />
```
Comment (satanmourner) [inline src/pages/signUp/email/index.tsx:17]: import from `/assets` please

File: src/pages/signUp/email/index.tsx [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+import css from './index.module.scss';
+import { useEmail } from './useEmail';
+
+export const Email = () => {
+  const { register, errors, handleSubmit, onContinue } = useEmail();
+  const { t } = useTranslation();
+
+  return (
+    <div className={css['container']}>
+      <div className={`w-full lg:min-w-[600px] ${css['formSection']}`}>
```
Comment (satanmourner) [inline src/pages/signUp/email/index.tsx:16]: follow BEM. use `form` instead

File: src/pages/signUp/email/index.tsx [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+                  {t('socious-login')}
+                </Button>
+                <Button
+                  color="primary"
+                  variant="outlined"
+                  // onClick={() => {
+                  //     tried();
+                  //     navigate(`/oauth/google${event.name && `?event_name=${event.name}`}`);
+                  // }}
+                  style={{ display: 'flex', gap: '12px' }}
```
Comment (satanmourner) [inline src/pages/signUp/email/index.tsx:50]: use `customStyle` prop instead please

File: src/pages/signUp/email/index.tsx [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+                register={register}
+                placeholder={t('email-placeholder')}
+                errors={errors['email']?.message ? [errors['email']?.message.toString()] : undefined}
+              />
+              <div className="flex flex-col gap-4">
+                <Button color="primary" onClick={handleSubmit(onContinue)}>
+                  {t('primary-btn-label')}
+                </Button>
+                <Button color="primary" variant="outlined" style={{ display: 'flex', gap: '12px' }}>
+                  <img src={'/images/logo/socious-logo.svg'} alt="" />
```
Comment (satanmourner) [inline src/pages/signUp/email/index.tsx:40]: import from `/assets` please

File: src/pages/signUp/profile/index.module.scss [hunk 20]
```diff
@@ -0,0 +1,18 @@
+.container {
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+    gap: 2rem;
+
+    @media (min-width: 640px) {
+        max-width: 480px;
+    }
```
Comment (satanmourner) [inline src/pages/signUp/profile/index.module.scss:9]: - `$md`
- rem please

**Review by satanmourner — CHANGES_REQUESTED** [review]

File: src/core/adaptors/media.ts [hunk 21]
```diff
@@ -0,0 +1,32 @@
+export interface UploadMediaAdaptorRes {
```
[the remark this answers is not in the document]
Comment (marjan-ara) [reply src/core/adaptors/media.ts:1]: @satanmourner let's discuss it after daily
I think in API we should define required types by server (both in request and response), in adaptor we can map our types to api types

[the remark this answers is not in the document]
Comment (satanmourner) [reply src/core/adaptors/media.ts:1]: whenever you change the adaptors as our convention, could you ping me to merge this PR? Thanks

**Review by satanmourner — APPROVED** [review]

**Pull request merged**